src/platform/packages/shared/kbn-lens-embeddable-utils/config_builder/schema/charts/xy.ts

```diff
@@ -0,0 +1,661 @@
+      schema.object(
+        {
+          ...sharedLegendSchema,
+          inside: schema.literal(true),
+          columns: schema.maybe(
+            schema.number({ min: 1, max: 5, meta: { description: 'Number of legend columns' } })
+          ),
+          alignment: schema.maybe(
+            schema.oneOf(
+              [
+                schema.literal('top_right'),
+                schema.literal('bottom_right'),
+                schema.literal('top_left'),
+                schema.literal('bottom_left'),
+              ],
+              { meta: { description: 'Legend alignment inside the chart' } }
+            )
+          ),
+        },
+        { meta: { description: 'Internal legend positioned inside the chart' } }
+      ),
+    ])
+  ),
+
+  fitting: schema.maybe(
+    schema.object(
+      {
+        type: schema.oneOf(
+          [
+            schema.literal('None'),
+            schema.literal('Zero'),
+            schema.literal('Linear'),
+            schema.literal('Carry'),
+            schema.literal('Lookahead'),
+            schema.literal('Average'),
+            schema.literal('Nearest'),
+          ],
+          { meta: { description: 'Fitting function type for missing data' } }
+        ),
+        dotted: schema.maybe(
+          schema.boolean({ meta: { description: 'Show fitted values as dotted lines' } })
+        ),
+        endValue: schema.maybe(
+          schema.oneOf([schema.literal('None'), schema.literal('Zero'), schema.literal('Nearest')])
+        ),
+      },
+      {
+        meta: {
+          description:
+            'Missing data interpolation configuration (only valid fitting types applied per chart type)',
+        },
+      }
+    )
+  ),
+  axis: schema.maybe(
+    schema.object(
+      {
+        x: schema.maybe(
+          schema.object(
+            {
+              ...sharedAxisSchema,
+              extent: schema.maybe(
+                schema.oneOf([
+                  schema.object(
+                    {
+                      type: schema.literal('full'),
+                      integer_rounding: schema.maybe(schema.boolean()),
+                    },
+                    { meta: { description: 'Full extent mode for X-axis' } }
+                  ),
+                  schema.object(
+                    {
+                      type: schema.literal('custom'),
+                      start: schema.number({ meta: { description: 'Custom X-axis start value' } }),
+                      end: schema.number({ meta: { description: 'Custom X-axis end value' } }),
+                      integer_rounding: schema.maybe(schema.boolean()),
+                    },
+                    { meta: { description: 'Custom X-axis extent' } }
+                  ),
+                ])
+              ),
+            },
+            { meta: { description: 'X-axis (horizontal) configuration' } }
+          )
+        ),
+        left: schema.maybe(yAxisSchema),
+        right: schema.maybe(yAxisSchema),
```

Comment:
I believe we need to reconsider these left/right and describe them more as `Y something`, like `primaryY` `secondaryY` or something like that. The reason is that the position only made sense with non rotated charts